src/main/java/de/azapps/kafkabackup/sink/BackupSinkTask.java

```diff
@@ -50,15 +71,49 @@ public void start(Map<String, String> props, OffsetSink overrideOffsetSink) {
             if(overrideOffsetSink != null) {
                 offsetSink = overrideOffsetSink;
             } else {
-                AdminClient adminClient = AdminClient.create(config.adminConfig());
+                adminClient = AdminClient.create(config.adminConfig());
```

Comment:
Would it make sense to use the adminClient in the EndOffsetReader instead creating an own client?

Reply:
Not sure that I follow, I am not aware that the AdminClient is capable of retrieving the end offsets of partitions.